Sentiment Analysis API Integration

eriyezab opened the pull request.

eriyezab committed:
Add dependecy to pom.xml to allow for use of Cloud Natural language API.

eriyezab committed:
Add score attribute to the Comment class to hold the sentiment score of the message.

eriyezab committed:
Fix spelling in comment for score attribute.

eriyezab committed:
Add sentiment analysis functionality to comments and display the sentiment score with the comment.

eriyezab committed:
Attend to todo to remove comment.

shrutip90 commented on portfolio/src/main/java/com/google/sps/data/Comment.java:
```diff
@@ -30,15 +30,22 @@
     this.timestamp = timestamp;
   }
   
   // TODO: Add getters for private attributes
```
You can move this TODO to class level comments.

shrutip90 commented on portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:
```diff
@@ -123,18 +129,36 @@ private Query makeQueryFromParams(HttpServletRequest request) {
+   * @param message The comment the user made that will be analyzed.
+   * @return The sentiment score that was given to the comment.
+   */
+  private float getSentimentScore(String message) throws IOException {
```
Looks like this service may throw exceptions. Can you add exception handling for this so that the page doesn't. crash in those cases, but. displays something else or ignores the results from this call.

shrutip90 commented on portfolio/src/main/java/com/google/sps/data/Comment.java:
```diff
@@ -30,15 +30,22 @@
   private String message;
 
+  /** The sentiment score of the message. */
+  private float score;
```
sentimentScore might be a clearer name. But I'll leave it up to you. This also works.

eriyezab committed:
Change attribute name to be more accurate.

eriyezab committed:
Handle exception thrown by sentiment analysis api and remove comments with low score.

shrutip90 commented on portfolio/src/main/webapp/comments.js:
```diff
@@ -17,7 +17,9 @@ function checkCommentPosted() {
   if(posted === "false") {
-    window.alert("Comment not posted!");
+    const reason = url.searchParams.get("reason");
+    const message = (reason === "score") ? "Your sentiment score was too low!" : "Your message was empty!";  
```
You can use a switch statement here, since you have separate reasons for both the conditions, to be more clear and to avoid breaking this code in case you add more reasons later.

shrutip90 commented on portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:
```diff
@@ -123,18 +136,42 @@ private Query makeQueryFromParams(HttpServletRequest request) {
+      languageService.close();
+      return score;
+    } catch (IOException e) {
+      System.out.println("This error ocurred getting the sentiment score:");
```
You can add the stack trace in the same log statement.

eriyezab committed:
Update DataServlet.java

Move stack trace into single log statement

eriyezab committed:
Update comments.js

Use switch statement instead of ternary operation for message and move into its own function for improved readability and in case more reasons are added in the future.
